src/kudu/consensus/consensus_peers.cc

```diff
@@ -616,6 +619,47 @@ void PeerProxyPool::Clear() {
   peer_proxy_map_.clear();
 }
 
+template <class RespType>
+void CheckAndEnforceResponseToken(const std::string& method_name,
+                                  RespType* response,
+                                  boost::optional<std::string> rpc_token,
+                                  bool enforce) {
```

Comment:
This class is on the outgoing request path and we're checking that the **response** returned to us has the right token.

CheckRaftRpcTokenOrRespond() is in the incoming request path where we check the **request** sent to us has the right token.

We need checks on both sides because both requests and responses trigger side effects. Responses for example can trigger term advancements and we need to reject them for bad instances.

Reply:
There are multiple kinds of error messages on requests. We don't have to send a term back on a bad request. We can just drop it. In that case response path should not have to do anything?